app/ai/voice/agents/breeze_buddy/agent/ivr.py

```
@@ -411,37 +417,37 @@ async def handle_ivr_menu(
 
-        # Generate cache key from ivr_greeting text and voice
-        # MD5 hash ensures constant key size regardless of input length
-        cache_components = f"greeting:{ivr_greeting}|voice:{voice_name}"
+        # Cache key includes voice provider+id so different configs get different audio
+        voice_key = (
+            f"{voice_config.provider.value}:{voice_config.voice_id}"
+            if voice_config
+            else "default"
+        )
+        cache_components = f"greeting:{ivr_greeting}|voice:{voice_key}"
         cache_key = f"{IVR_AUDIO_CACHE_PREFIX}{hashlib.md5(cache_components.encode()).hexdigest()}"
```

Ruff (0.15.6)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)